codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/HttpBoundProtocolPayloadGenerator.kt

```diff
@@ -164,6 +169,19 @@ class HttpBoundProtocolPayloadGenerator(
         }
     }
 
+    private fun generateSigner(): Writable {
+        when (target) {
+            CodegenTarget.CLIENT -> {
+                if (codegenContext.serviceShape.hasTrait<SigV4Trait>()) {
+                    return writable { rust("_config.new_event_stream_signer(properties.clone())") }
+                }
+                return writable { rust("_config.new_event_stream_no_op_signer(properties.clone())") }
```

Comment:
The benefit of having one `new_event_stream_signer` function on the config is that you won't need this if/else for each signing trait.